src/predictor/gpu_predictor.cu

```diff
@@ -375,7 +400,6 @@ class GPUPredictor : public xgboost::Predictor {
     size_t n_classes = model.param.num_output_group;
     size_t n = n_classes * info.num_row_;
     const HostDeviceVector<bst_float>& base_margin = info.base_margin_;
-    out_preds->Shard(GPUDistribution::Granular(devices_, n_classes));
```

Comment:
If `out_preds` are not sharded, copying/filling will take place on host. This is something we'd like to avoid, at least when not using external memory.

Reply:
Added back.